### ISSUE TITLE
The filter products by attribute block is not using the product attributes lookup table

### ISSUE THREAD

**Issue by Konamiman** [woothemes/woocommerce#42600]
## Describe the bug

WooCommerce 6.3 introduced [a new product attributes lookup table](https://developer.woocommerce.com/2022/02/02/new-product-filtering-by-attributes-rolling-out-in-woocommerce-6-3/) whose usage can be enabled and disabled by a setting. The filter by attribute block is not using it, cuasing discrepancies between the product counts returned by the block and by the old WooCommerce "filter by attribute" widget when the lookup table is in use.

## To reproduce & Expected behavior

See the testing instructions of https://github.com/woocommerce/woocommerce/pull/34432 for the setup (no need to actually use non-ASCII attributes), these use the old WooCommerce widget and show the expected behavior; then see [this comment](https://github.com/woocommerce/woocommerce/pull/34432#issuecomment-1225464870). The filtering problem is handled in [a separate issue](https://github.com/woocommerce/woocommerce-blocks/issues/6993).

## Screenshots

See the different product counts when applying the above testing instructions (the correct ones are the "(1) - (1)"):

![image](https://user-images.githubusercontent.com/937723/186695539-a0ae74e2-65f5-4b66-988e-7e722c232a65.png)

### WordPress

-   Core version: 6.0
-   WooCommerce version: 6.9
-   WooCommerce Blocks version: 8.3.1

## Additional context

The proper way to get the product counts is to run `wc_get_container()->get( Filterer::class )->get_filtered_term_product_counts` (see `WC_Widget_Layered_Nav::get_filtered_term_product_counts`, this will already take care of whether the lookup table usage is enabled or not).

**Comment by gigitux** [comment]
@dinhtungdu Based on my memory and a code check, it looks like we're using the product attributes lookup table:  
<https://github.com/woocommerce/woocommerce/blob/45b7639f809afaa467591164f724aab91c671a4c/plugins/woocommerce/src/Blocks/QueryFilters.php#L398-L400>  

Do you think that can we close this issue?

**Comment by dinhtungdu** [comment]
@gigitux Good call! I can confirm that, both current and new filter blocks use look up table. I'm closing this one.
